tests/integration/test_deploy.py

```diff
@@ -1,41 +1,59 @@
 from pathlib import Path
 import logging
-
 import pytest
 from lightkube import Client, codecs
 from lightkube.resources.core_v1 import Pod, Service
 import yaml
+from tenacity import retry, wait_exponential, stop_after_delay, before_log
 
 CHARM_DIR = Path(__file__).parent.parent.parent.resolve()
-SPEC_FILE = Path(__file__).parent / "validate-dns-spec.yaml"
+SPEC_FILE = Path(__file__).parent / "data/validate-dns-spec.yaml"
```

Comment:
for the poor sole trying to run this on Windows(TM)
```suggestion
SPEC_FILE = Path(__file__).parent / "data" / "validate-dns-spec.yaml"
```